elide-core/src/main/java/com/yahoo/elide/security/executors/ActivePermissionExecutor.java

```diff
@@ -268,11 +324,16 @@ public void executeCommitChecks() {
      *
      * @param expressions expressions to execute
      */
-    private void executeExpressions(final Expressions expressions,
+    private ExpressionResult executeExpressions(final Expressions expressions,
                                     final Class<? extends Annotation> annotationClass) {
         Expression expression = expressions.getOperationExpression();
         ExpressionResult result = expression.evaluate();
-        if (result == DEFERRED) {
+
+        // Record the check
+        Long checkOccurrences = requestScope.getCheckStats().getOrDefault(expression.toString(), 0L) + 1;
```

Comment:
Instead of adding getters, why not just add a single method to the PermissionExecutor to check whether or not to short circuit.  This also hides the implementation details from PersistentResource.